### PR TITLE
JAPI-273: new searchFiles method

### PR DESCRIPTION
**Pull request by drealeed**
Submitting this as it's needed for HIPIE capabilities.

Tested with unit tests, also tested end-to-end by pointing HIPIE at the new method and running tests.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 1]
```diff
@@ -953,12 +953,30 @@ public void deleteSuperFileSubfiles(String superfile, List<String> subfiles) thr
...
+    }
+    /**
+     * searchFiles
+     *
+     * @param logicalFilename
+     *            - the filename to search for
+     * @param cluster
+     *            - the cluster to search on
+     * @param pagesize
+     *            - the page to begin on
```
Comment (Michael-Gardner) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: looks like the javadoc params might be incorrect.  I think 'pageStartFrom' should probably say what 'pagesize' is labeled.

**Comment by drealeed** [comment]
@rpastrana, specifically what branch should I pull to?  Candidate 7.6.24?

Also, because the testing of searchFiles() will need to be remote test (with an hpcc system available), which cluster do you want me to point to for the unit test?

**Comment by rpastrana** [comment]
@drealeed target 7.6.x. the next rc will be cut from that branch tomorrow.
BaseRemoteTest has a thorcluster member var you can use, it's populated from system prop "thorcluster", and defaults to "mythor". 
If that doesn't fit your criteria, a custom system property could be set within the dfutest client code

**Comment by drealeed** [comment]
unit test added and tested. I @Ignored it for now so it doesn't break the automated tests until I figure out what data to use on that cluster.

**Pull request merged**